samcli/cli/command.py

```diff
@@ -1,12 +1,13 @@
 """
```

Comment:
hmmm this PR is labeled as a `design` but it seems like we're mixing in some implementation changes. Should we have those in a separate PR?

Reply:
I added to showcase the poc that it can be done, more than happy to split it out.